akka-actor/src/main/scala/akka/io/dns/internal/AsyncDnsResolver.scala

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (C) 2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.io.dns.internal
+
+import akka.actor.{ Actor, ActorLogging, ActorRef, Props }
+import akka.annotation.InternalApi
+import akka.io.dns.DnsProtocol.{ Ip, RequestType, Srv }
+import akka.io.dns.internal.DnsClient._
+import akka.io.dns.{ DnsProtocol, DnsResourceRecord, DnsSettings }
+import akka.pattern.{ ask, pipe }
+import akka.util.Timeout
+import com.typesafe.config.Config
+
+import scala.collection.immutable.Seq
+import scala.collection.{ breakOut, immutable }
+import scala.concurrent.Future
+import scala.concurrent.forkjoin.ThreadLocalRandom
+
+private[io] final class AsyncDnsResolver(cache: AsyncDnsCache, config: Config) extends Actor with ActorLogging {
+
+  import AsyncDnsResolver._
+
+  implicit val ec = context.dispatcher
+  val settings = new DnsSettings(config)
+
+  // For ask to DNS Client
+  implicit val timeout = Timeout(settings.ResolveTimeout)
+
+  val nameServers = settings.NameServers
+
+  log.debug("Using name servers [{}]", nameServers)
+
+  private var requestId: Short = 0
+  private def nextId(): Short = {
+    requestId = (requestId + 2).toShort
+    requestId
+  }
+
+  private val resolvers: IndexedSeq[ActorRef] = nameServers.map({ ns ⇒
+    context.actorOf(Props(classOf[DnsClient], ns))
+  })(breakOut)
+
+  override def receive: Receive = {
+    case DnsProtocol.Resolve(name, _) if isInetAddress(name) ⇒
+      log.warning("Tried to resolve ip {}. Ignoring.", name)
+    case DnsProtocol.Resolve(name, mode) ⇒
+      resolve(name, mode) pipeTo sender()
+  }
+
+  def next = resolvers(ThreadLocalRandom.current().nextInt(resolvers.length))
```

Comment:
AFAIR this is undesirable and they should be tried in order btw, see ticket on akka-dns